### ISSUE TITLE
Node 0.10 compatibility and migration

### ISSUE THREAD

**Issue by balupton** [docpad/docpad#1008]
DocPad should exit on node 0.10 and inform users to upgrade to Node 0.10 compat 

[Node v4 is around the corner](https://medium.com/node-js-javascript/4-0-is-the-new-1-0-386597a3436d) and node 0.10 is years old, slow, and buggy. It's time we turn older environments into fatal errors to prevent issues arising. To upgrade, see our [node install guide](https://learn.bevry.me/node/install).

Issues relating to node 0.10:
- https://github.com/docpad/docpad/issues/1007
- https://github.com/bevry/lazy-require/issues/6

**Comment by balupton** [comment]
The next release of DocPad will include the babel polyfill to work around issues like this, work was commenced over at https://github.com/docpad/docpad/issues/1045 for this

For those on node 0.10, is there any specific reasons you are on node 0.10, or tied to it? Node 0.10 is more than 3 years old now: https://github.com/nodejs/node/releases/tag/v0.10.0

In the mean time, node 0.10 users could perhaps do this:

```
npm install -g babel-cli
babel-node ./node_modules/.bin/docpad run
```

**Comment by balupton** [comment]
So this landed with v6.55.6:
https://github.com/docpad/docpad/commit/80c27f7d01c6f7a65c0061aa48504afe8453a1dd

Apparently related to  #725
